pcf/particle/aws/ecr/ecr_repository.py

```diff
@@ -0,0 +1,142 @@
+# Copyright 2018 Capital One Services, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from pcf.core import State
+from pcf.util import pcf_util
+from pcf.core.aws_resource import AWSResource
+from pcf.core.pcf_exceptions import TooManyResourceException
+import logging
+
+logger = logging.getLogger(__name__)
+
+class ECRRepository(AWSResource):
+    """
+    This is the implementation of Amazon's EC2 Container Registry (ECR) Repository.
+    """
+
+    flavor = "ecr_repository"
+
+    state_lookup = {
+        "available": State.running,
+        "pending": State.pending,
+        "missing": State.terminated
+    }
+
+    equivalent_states = {
+        State.running: 1,
+        State.stopped: 0,
+        State.terminated: 0
+    }
+
+    DEFINITION_FILTER = {
+        'repositoryName',
```

Comment:
can they add other fields to the desired def that cannt go in start. if not they we dont need this and can just pass in the full definition.

Reply:
Short answer is No. But the describe API call response we use for current definition has more than just the repositoryName. This filter is actually more for current def than desired def. Although I propose that we apply a filter both current and desired def regardless. It'll guarantee that just in case AWS decides to change their API response to provide more fields and it won't break the comparison.